File: q0435/q0435.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));
        int res = 0;
        int prevEndTime = intervals[0][1];
        for (int i = 1; i < intervals.length; i++) {
            int newStartTime = intervals[i][0];
            int newEndTime = intervals[i][1];
            if (newStartTime >= prevEndTime) {
                prevEndTime = newEndTime;
            } else {
                res++;
            }
        }
        return res;
    }
}

File: q0435/q0435.cpp
class Solution
{
    static bool endTimeComparator(vector<int> &i1, vector<int> &i2)
    {
        return i1[1] < i2[1];
    }

public:
    int eraseOverlapIntervals(vector<vector<int>> &intervals)
    {
        sort(intervals.begin(), intervals.end(), endTimeComparator);
        int res = 0;
        int prevEndTime = INT_MIN;
        for (const auto &i : intervals)
        {
            int newStartTime = i[0];
            int newEndTime = i[1];
            if (newStartTime >= prevEndTime)
            {
                prevEndTime = newEndTime;
            }
            else
            {
                res++;
            }
        }
        return res;
    }
};